execution/gethexec/executionengine.go

```diff
@@ -1082,8 +1085,52 @@ func (s *ExecutionEngine) Start(ctx_in context.Context) {
 	}
 }
 
-func (s *ExecutionEngine) Maintenance(capLimit uint64) error {
-	s.createBlocksMutex.Lock()
-	defer s.createBlocksMutex.Unlock()
-	return s.bc.FlushTrieDB(common.StorageSize(capLimit))
+func (s *ExecutionEngine) ShouldTriggerMaintenance(trieLimitBeforeFlushMaintenance time.Duration) bool {
+	if s.runningMaintenance.Load() {
+		return false
+	}
+
+	if trieLimitBeforeFlushMaintenance == 0 {
+		return true
```

Comment:
default should be to always return false

Reply:
Forgot to change this after I addressed your comments to change the maintenance triggering behavior, thanks :)